### PR TITLE
[CARBONDATA-1158] Hive integration code optimization

### PR DESCRIPTION
**Pull request by chenliang613**
Hive integration code optimization:
1. Remove redundant and unused code.
2. Optimize some code:
   a) Convert some internal functions from public to private.
   b) Fix some code which may generate error.
   c) Change code as per java code style.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/275/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2396/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2543/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/434/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2557/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/449/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/449/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/449/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.dataretention/DataRetentionConcurrencyTestCase/DataRetention_Concurrency_load_date/'><strong>org.apache.carbondata.spark.testsuite.dataretention.DataRetentionConcurrencyTestCase.DataRetention_Concurrency_load_date</strong></a></li></ul>

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2570/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/464/

File: integration/hive/src/main/java/org/apache/carbondata/hive/CarbonArrayInspector.java [hunk 1]
```diff
@@ -29,31 +30,27 @@
  * The CarbonHiveArrayInspector will inspect an ArrayWritable, considering it as an Hive array.
  * It can also inspect a List if Hive decides to inspect the result of an inspection.
  */
-public class CarbonArrayInspector implements SettableListObjectInspector {
+class CarbonArrayInspector implements SettableListObjectInspector {
 
-  private ObjectInspector arrayElementInspector;
+  private final ObjectInspector arrayElementInspector;
 
   public CarbonArrayInspector(final ObjectInspector arrayElementInspector) {
```
Comment (anubhav100) [inline integration/hive/src/main/java/org/apache/carbondata/hive/CarbonArrayInspector.java]: can remove public access specifier

File: integration/hive/src/main/java/org/apache/carbondata/hive/CarbonHiveSerDe.java [hunk 2]
```diff
@@ -53,20 +53,17 @@
...
 import org.apache.hadoop.io.Writable;
 
-
 /**
  * A serde class for Carbondata.
  * It transparently passes the object to/from the Carbon file reader/writer.
  */
-@SerDeSpec(schemaProps = {serdeConstants.LIST_COLUMNS, serdeConstants.LIST_COLUMN_TYPES})
+@SerDeSpec(schemaProps = { serdeConstants.LIST_COLUMNS, serdeConstants.LIST_COLUMN_TYPES })
 public class CarbonHiveSerDe extends AbstractSerDe {
```
Comment (anubhav100) [inline integration/hive/src/main/java/org/apache/carbondata/hive/CarbonHiveSerDe.java]: @chenliang613 can remove public access specifier

File: integration/hive/src/main/java/org/apache/carbondata/hive/CarbonObjectInspector.java [hunk 3]
```diff
@@ -34,18 +33,16 @@
...
 
-public class CarbonObjectInspector extends SettableStructObjectInspector {
+class CarbonObjectInspector extends SettableStructObjectInspector {
   private final TypeInfo typeInfo;
-  private final List<TypeInfo> fieldInfos;
-  private final List<String> fieldNames;
   private final List<StructField> fields;
   private final HashMap<String, StructFieldImpl> fieldsByName;
 
   public CarbonObjectInspector(final StructTypeInfo rowTypeInfo) {
```
Comment (anubhav100) [inline integration/hive/src/main/java/org/apache/carbondata/hive/CarbonObjectInspector.java]: @chenliang613 can remove public access specifier

File: integration/hive/src/main/java/org/apache/carbondata/hive/CarbonObjectInspector.java [hunk 4]
```diff
@@ -198,27 +178,23 @@ public StructFieldImpl(final String name, final ObjectInspector inspector, final
...
       return "";
     }
 
-    @Override
-    public String getFieldName() {
+    @Override public String getFieldName() {
       return name;
     }
 
     public int getIndex() {
```
Comment (anubhav100) [inline integration/hive/src/main/java/org/apache/carbondata/hive/CarbonObjectInspector.java]: @chenliang613 can remove public acess specifier

**Comment by anubhav100** [comment]
@chenliang613 i reviewed the pr its  working fine on my pc,just removed the some unnecessary public access specifier as suggested by me in comments every thing else looks good to me

**Comment by chenliang613** [comment]
@anubhav100  thanks , fix comments

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2627/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/534/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**